portfolio/src/main/webapp/script.js

```diff
@@ -4,75 +4,85 @@ $(document).ready(function() {
 
-        
     const fact = facts[Math.floor(Math.random() * facts.length)];
 
     const factContainer = document.getElementById('fact-container');
     factContainer.innerText = fact;
 }
 
 //Typewriting Effect
-var TxtType = function(el, toRotate, period) {
-var TxtType = function(el, toRotate, period) {
+{
+    //call new startRotation(element, data, period) in the window.onload function.
+
+
+    //  * @param {any} element This parameter is always needed.
+    //  * @param {any} toRotate This parameter is always needed.
+    //  * @param {number} period This paramter is always needed.
+
+    let textBeingTyped = function(element, toRotate, period) {
         this.toRotate = toRotate;
-        this.el = el;
+        this.element = element;
         this.loopNum = 0;
         this.period = parseInt(period, 10) || 2000;
-        this.txt = '';
+        this.text = '';
         this.tick();
         this.isDeleting = false;
     };
 
-    TxtType.prototype.tick = function() {
+    textBeingTyped.prototype.tick = function() {
         var i = this.loopNum % this.toRotate.length;
         var fullTxt = this.toRotate[i];
 
         if (this.isDeleting) {
-        this.txt = fullTxt.substring(0, this.txt.length - 1);
+            this.text = fullTxt.substring(0, this.text.length - 1);
         } else {
-        this.txt = fullTxt.substring(0, this.txt.length + 1);
+            this.text = fullTxt.substring(0, this.text.length + 1);
         }
 
-        this.el.innerHTML = '<span class="wrap">'+this.txt+'</span>';
+        this.element.innerHTML = '<span class="wrap">' + this.text + '</span>';
 
         var that = this;
         var delta = 200 - Math.random() * 100;
 
         if (this.isDeleting) { delta /= 2; }
 
         if (!this.isDeleting && this.txt === fullTxt) {
-        delta = this.period;
-        this.isDeleting = true;
+            delta = this.period;
+            this.isDeleting = true;
         } else if (this.isDeleting && this.txt === '') {
-        this.isDeleting = false;
-        this.loopNum++;
-        delta = 500;
+            this.isDeleting = false;
+            this.loopNum++;
+            delta = 500;
         }
 
         setTimeout(function() {
-        that.tick();
+            that.tick();
         }, delta);
     };
 
     window.onload = function() {
         var elements = document.getElementsByClassName('typewrite');
-        for (var i=0; i<elements.length; i++) {
-            var toRotate = elements[i].getAttribute('data-type');
-            var period = elements[i].getAttribute('data-period');
+        for (const element of elements) {
+            var toRotate = element.getAttribute('data-type');
+            var period = element.getAttribute('data-period');
             if (toRotate) {
-              new TxtType(elements[i], JSON.parse(toRotate), period);
+                new textBeingTyped(element, JSON.parse(toRotate), period);
             }
+            element++;
```

Comment:
can you remove element++? the for loop will iterate over any other 'typewrite' elements